src/app/authentication/auth.service.ts

```diff
@@ -1,46 +1,81 @@
 import { Injectable } from '@angular/core';
+import * as JWT from 'jwt-decode';
 import { AppComponent } from '../app.component';
 
 declare let Msal: any;
+
 @Injectable()
 export class AuthService {
+
     private app: any;
 
     constructor() {
-        this.app = new Msal.UserAgentApplication((window as any).ClientId, '', this.authCallback, {});
-    }
-
-    public authCallback = (errorDesc, token, error) => {
-        if (token) {
-            localStorage.setItem('status', 'authenticated');
-        } else if (errorDesc || error) {
-            localStorage.setItem('status', 'anonymous');
-        }
+        this.app = new Msal.UserAgentApplication((window as any).ClientId, '',
+            {});
     }
 
     public async login() {
-        return this.app.loginRedirect(this.getScopes());
+        return this.app.loginPopup(this.defaultUserScopes())
+            .then(async () => {
+                return this.getTokenSilent();
+            }, () => {
+                return false;
+            });
     }
 
     public logout() {
         this.app.logout();
     }
 
-    public getToken() {
-        return this.app.acquireTokenSilent(this.getScopes())
+    public getTokenSilent(scopes: any = []) {
```

Comment:
Use the type for scopes instead of any

Reply:
This function is also returning a value, specify the return type of the function.

Reply:
I don't get what you mean

Reply:
@thewahome 

In the base.ts you will see the type of a permission, `IPermissionScope`

For the second point, if we have a function like the one below returning a value, we should specify the return type in the function signature.
```
public returnSomething() : string {
   return 'something';
}
```